From ce36e47b2e83957097b57396f0f3f44c14477c5b Mon Sep 17 00:00:00 2001
From: Kemal Rasim Sh <kshakir@codeaurora.org>
Date: Thu, 29 Oct 2020 16:05:13 +0200
Subject: [PATCH 1/1] Unblock GSTD pipeline if a plugin refuses to change its
 state

If a plugin gets unexpectedly stopped, the pipe will get blocked
in an unrecoverable condition. To prevent the blocking, this change allows
the pipe to be deleted if the state update fails.

Signed-off-by: Kemal Rasim Sh <kshakir@codeaurora.org>
---
 gstd/gstd_pipeline_deleter.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/gstd/gstd_pipeline_deleter.c b/gstd/gstd_pipeline_deleter.c
index 6aed1db..4cac3cd 100644
--- a/gstd/gstd_pipeline_deleter.c
+++ b/gstd/gstd_pipeline_deleter.c
@@ -96,9 +96,8 @@ gstd_pipeline_deleter_delete (GstdIDeleter * iface, GstdObject * object)
   if (ret)
     return ret;
 
-  ret = gstd_object_update (state, "NULL");
-  if (ret)
-    return ret;
+  /* Delete the object even it fails */
+  gstd_object_update (state, "NULL");
 
   g_object_unref (state);
   g_object_unref (object);
--
2.17.1
